ci2/ci/github.py

```diff
@@ -190,13 +189,45 @@ def __init__(self, branch):
 
         branch_gh_json = await gh.getitem(f'/repos/{repo_ss}/git/refs/heads/{self.branch.name}')
         new_sha = branch_gh_json['object']['sha']
+        if new_sha != self.sha:
+            self.sha = new_sha
+            if self.prs:
+                for pr in self.prs:
+                    pr.batch = None
+                    pr.passing = None
 
         new_prs = {}
         async for gh_json_pr in gh.getiter(f'/repos/{repo_ss}/pulls?state=open&base={self.branch.name}'):
             number = gh_json_pr['number']
-            pr = PR.from_gh_json(gh_json_pr)
-            await pr.refresh(gh)
+            if self.prs is not None and number in self.prs:
+                pr = self.prs[number]
+                pr.update_from_gh_json(gh_json_pr)
+            else:
+                pr = PR.from_gh_json(gh_json_pr, self)
             new_prs[number] = pr
-
-        self.sha = new_sha
         self.prs = new_prs
+
+        for pr in self.prs.values():
+            await pr.refresh_review_state(gh)
```

Comment:
ok, I think there's still one missing piece of logic: where do we remove a pr that has been closed? It seems like we should track the numbers we saw in the `async for` loop.  After the loop, we should:
```
for k in self.prs.keys() - seen_pr_numbers:
    del self.prs[k]
```

Reply:
I reassign prs just before this:

 >         self.prs = new_prs

Reply:
👍